WPFHospitalEditor/AppointmentAnalysisWindow.xaml.cs

```diff
@@ -0,0 +1,94 @@
+        }
+
+        private void FillObservableCollection(int specialtyId)
+        {
+            examinationWithAvailableRescheduling = new ObservableCollection<ExaminationWithAvailableRescheduling>();
+            List<Examination> examinations = examinationServerController.GetBySpecialtyId(specialtyId).ToList();
+            foreach (Examination exam in examinations) {
+                ExaminationWithAvailableRescheduling examination = new ExaminationWithAvailableRescheduling(exam.Id,exam.PatientId, exam.DoctorId, exam.Priority, exam.TimeInterval.Start, DateTime.Now, exam.RequiredSpecialtyId);
+                examinationWithAvailableRescheduling.Add(examination);
+                examinationSearchComboBox.Items.Add(exam.Id);
+            }
+
+            foreach (ExaminationWithAvailableRescheduling exam in examinationWithAvailableRescheduling) {
+                RecommendationRequestDto recommendationRequestDto = new RecommendationRequestDto {
+                    DoctorId = exam.DoctorId,
+                    Preference = RecommendationPreference.Time,
+                    SpecialtyId = exam.RequiredSpecialtyId,
+                    TimeInterval = new TimeInterval(DateTime.Now.AddHours(1), DateTime.Now.AddDays(5))
+                };
+                List<RecommendationDto> recommendationDtos = schedulingController.GetAppointments(recommendationRequestDto);
+                exam.ReschedulingDate = recommendationDtos[0].TimeInterval.Start;
+            }
+        }
+
+        private void Close_Click(object sender, RoutedEventArgs e)
+        {
+            this.Close();
+        }
+        private void Cancel_Examination(object sender, RoutedEventArgs e)
+        {
+            int examinationId = int.Parse(examinationSearchComboBox.Text);
+            string cancel = examinationServerController.Cancel(examinationId);
+            if (!cancel.Equals("BadRequest"))
+            {
+                MessageBox.Show("Examination is successfuly CANCELED!");
+            }
+            else {
+                MessageBox.Show("Examination cannot be CANCELED!");
+            }
+            ExaminationWithAvailableRescheduling examForScheduling = null;
+            foreach (ExaminationWithAvailableRescheduling exam in examinationWithAvailableRescheduling) {
+                if (exam.ExaminationId == examinationId) {
+                    examForScheduling = exam;
+                }
+            }
+            RescheduleExamination(examForScheduling);
+        }
+
+        private void RescheduleExamination(ExaminationWithAvailableRescheduling exam) {
+            Examination examination = examinationServerController.ScheduleEmergencyExamination(exam.ReschedulingDate, exam.DoctorId, exam.PatientId);
+            if (examination != null)
+            {
+                MessageBox.Show("Examination is successfuly SCHEDULED!");
+            }
+            else {
+                MessageBox.Show("Examination is successfuly SCHEDULED!");
```

Comment:
It will be always successfully (typo in you're case ). Message are same in both cases.

Reply:
Fixed